### DISCUSSION TITLE
Should the default docker compose setup contain redis?

### DISCUSSION THREAD

**Discussion by stoically** [Netflix/conductor#3267] [Q&A] [answered]
**Describe the bug**
When cloning the repo, checking out the latest release tag and running `docker compose build` + `docker compose up -d` in the `docker` directory the conductor-server first starts, but then starts to spam hundreds of thousands of debug lines containing

> 639708 [scheduled-task-pool-2] WARN  com.netflix.dyno.queues.redis.RedisDynoQueue [] - cannot add 25e151ee-ed53-4d5e-9258-fd805842fa04 to the unack shard conductor_queues.test.UNACK._deciderQueue.c

https://conductor.netflix.com/gettingstarted/docker.html did not mention redis, but going by the debug message I assume redis is needed. Maybe it'd make sense to add redis to the default setup?

**Details**
Conductor version: 3.11.3
Persistence implementation: Default docker compose
Queue implementation: Default docker compose
Lock: None
Workflow definition:
Task definition:
Event handler definition:

**To Reproduce**
1. Clone repo
2. `cd docker`
3. `docker compose build`
4. `docker compose up -d`

**Comment by stoically** [comment, marked as answer]
I guess this should've been actually a discussion thread, not a bug issue.

**Reply by aravindanr** [reply]
In the default configuration, Conductor starts with an in-memory Redis implementation. The in-memory implementation is not meant for production use but allows for easier prototyping. Please ignore the WARN message, it does not signify any system degradation.

**Reply by stoically** [reply]
Thanks for elaborating. Then I guess I did something else wrong while trying to run a workflow, because it seemed that it isn't picked up.

Also, the WARN message was constantly logged and quickly was in the hundreds of thousands of lines. But maybe I did something wrong while setting it up.

**Comment by ropereraLK** [comment]
@aravindanr: I am using conductor only for prototyping. However, when I am trying to set up conductor on my local PC (Windows OS)  I am getting the same warn log in an endless loop and the execution of the application does not proceed after that.??

Please help on this..?

**Comment by ropereraLK** [comment]
> **Describe the bug** When cloning the repo, checking out the latest release tag and running `docker compose build` + `docker compose up -d` in the `docker` directory the conductor-server first starts, but then starts to spam hundreds of thousands of debug lines containing
> 
> > 639708 [scheduled-task-pool-2] WARN  com.netflix.dyno.queues.redis.RedisDynoQueue [] - cannot add 25e151ee-ed53-4d5e-9258-fd805842fa04 to the unack shard conductor_queues.test.UNACK._deciderQueue.c
> 
> https://conductor.netflix.com/gettingstarted/docker.html did not mention redis, but going by the debug message I assume redis is needed. Maybe it'd make sense to add redis to the default setup?
> 
> **Details** Conductor version: 3.11.3 Persistence implementation: Default docker compose Queue implementation: Default docker compose Lock: None Workflow definition: Task definition: Event handler definition:
> 
> **To Reproduce** Steps to reproduce the behavior:
> 
> 1. Clone repo
> 2. `cd docker`
> 3. `docker compose build`
> 4. `docker compose up -d`

@stoically : were you able to fix this issue after some workaround.?
